src/keydra/clients/splunk.py

```diff
@@ -245,74 +250,184 @@ def rotate_hectoken(self, inputname):
-                )
+    def _delete_splunkcloud_httpinput(self, inputname):
+        '''
+        Delete a Splunk HEC token on Splunk Cloud (Classic)
+
+        :param inputname: The name of the HTTP input
+        :type inputname: :class:`string`
+
+        :returns: Deployment task Id
+        :rtype: :class:`string`
+
+        '''
+        try:
+            self._service.delete(
+                '/services/dmc/config/inputs/__indexers/http/{}'.format(inputname),
+                output_mode='json'
             )
+        except HTTPError:
+            raise Exception('Error sending HEC token delete request')
+
+        return self._get_last_splunkcloud_deploytask()
+
+    def _create_splunkcloud_httpinput(self, inputname, inputconfig):
+        '''
+        Create a Splunk HEC input on Splunk Cloud (Classic)
+
+        :param inputname: The name of the HTTP input
+        :type inputname: :class:`string`
+        :param inputconfig: The configuration of the new HEC Input
+        :type inputconfig: :class:`dict`
+
+        :returns: Tuple of Deployment task Id and deployed configuration
+        :rtype: :class:`tuple`
+
+        '''
 
         try:
             createresp = self._service.post(
                 '/services/dmc/config/inputs/__indexers/http',
-                **config
+                headers=[('Content-Type', 'application/json')],
+                output_mode='json',
+                body=json.dumps(inputconfig)
             )
-            newconfig = json.loads(createresp['body'].read())['entry'][0]
-
-            return newconfig['content']['token']
 
-        except Exception as e:
+        except HTTPError as error:
             raise Exception(
-                'Error creating new HEC token {} on Splunk host '
-                '{}. Response: {}'.format(
-                    inputname,
-                    self._service.host,
-                    createresp['body'].read()
-                )
-            ) from e
+                'Error creating input: {}'.format(error)
+            )
+
+        response = json.loads(createresp['body'].read())
+
+        return self._get_last_splunkcloud_deploytask(), response['entry'][0]
+
+    def rotate_hectoken_cloud(self, inputname):
+        '''
+        Rotate a Splunk HEC token on Splunk Cloud (Classic)
+
+        Unfortunately you can rotate in place on Splunk Cloud, so we have
+        to delete and recreate the HTTP input in order to get a new token.
+
+        https://docs.splunk.com/Documentation/SplunkCloud/8.2.2105/Admin/ManageHECtokens
+
+        :param inputname: The name of the HTTP input
+        :type inputname: :class:`string`
+
+        :returns: New token value if successful
+        :rtype: :class:`string`
+
+        '''
+        LOGGER.debug('Getting input details for {}'.format(inputname))
+
+        inputs = self._get_splunkcloud_httpinput(inputname)
+
+        if len(inputs) == 0:
+            raise Exception('Input was not found!')
```

Comment:
May want to output the input name, as debug level is usually disabled

Reply:
Updated!